File: new-algorithm/src/javaalgo/programmers/level1_제일_작은_수_제거하기.java
package javaalgo.programmers;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class level1_제일_작은_수_제거하기 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        int[] arr = new int[N];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        System.out.println(Arrays.toString(solution(arr)));
    }

    public static int[] solution(int[] arr) {

        if (arr.length == 1) {
            arr[0] = -1;
        } else {
            int removeElem = Arrays.stream(arr).min().getAsInt();

            arr = Arrays.stream(arr)
                    .filter(i -> i != removeElem)
                    .toArray();
        }

        return arr;
    }
}

File: new-algorithm/src/kotlinalgo/programmers/level1_제일_작은_수_제거하기.kt
package kotlinalgo.programmers

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val N = Integer.parseInt(readLine())
    val arr = IntArray(N)
    val st = StringTokenizer(readLine())
    for (i in 0 until  N) {
        arr[i] = Integer.parseInt(st.nextToken())
    }
    solutionRemove(arr).forEach { println(it) }
}

fun solutionRemove(arr: IntArray): IntArray =
    if (arr.size == 1) {
        intArrayOf(-1)
    } else {
        val min = arr.toList().minOrNull()
        arr.filter { it != min }.toIntArray()
    }

